src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -97,6 +98,11 @@ internal partial class Binder
                 return CreateTupleLiteralConversion(syntax, (BoundTupleLiteral)source, conversion, isCast, destination, diagnostics);
             }
 
+            if (conversion.Kind == ConversionKind.ImplicitDefault)
+            {
+                return new BoundDefaultOperator(source.Syntax, destination);
+            }
```

Comment:
You appear to be throwing away the default operator and the conversion, and replacing it with its lowering here. The design pattern is to do binding only in the binder, and lowering in the local lowering phase. This pattern will make it difficult to get the correct result in the semantic model.